src/network/taraxa_capability.cpp

```diff
@@ -546,21 +561,26 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
           LOG(log_wr_dag_sync_) << "DagBlockValidation failed " << status.second;
           status.second.push_back(block.getHash());
           requestBlocks(_nodeID, status.second);
-          if (iBlock + transactionCount + 1 >= itemCount) break;
           continue;
         }
 
-        LOG(log_nf_dag_sync_) << "Storing block " << block.getHash().toString() << " with " << newTransactions.size()
+        LOG(log_dg_dag_sync_) << "Storing block " << block.getHash().toString() << " with " << newTransactions.size()
                               << " transactions";
         if (block.getLevel() > peer->dag_level_) peer->dag_level_ = block.getLevel();
         dag_blk_mgr_->insertBroadcastedBlockWithTransactions(block, newTransactions);
+      }
 
-        if (iBlock + transactionCount + 1 >= itemCount) break;
+      if (is_final_sync_packet) {
+        syncing_state_.set_dag_syncing(false);
+        LOG(log_nf_dag_sync_) << "Received final DagBlocksSyncPacket with blocks: " << received_dag_blocks_str;
+      } else {
+        syncing_state_.set_last_sync_packet_time();
```

Comment:
I guess I would double check where we call this and if we are sure we call it only on places where we should

Reply:
it is called when syncing packet is received and it passes all check. And the other palace is when we are setting syncing to true